index.js

```diff
@@ -1,38 +1,76 @@
+
 // create a new Discord client
 const client = new Client({ intents: [ Intents.FLAGS.GUILDS,  Intents.FLAGS.GUILD_MESSAGES ] });
 const regex = /\[\[([^\[][^\]]*)\]\]/g;
 const MAX_RESULTS = 25;
+const MAX_RESPONSES = 3;
 
 // when the client is ready, run this code
 // this event will only trigger one time after logging in
 client.once('ready', async () => {
     this.gallery = new Gallery();
     await this.gallery.parseGallery(process.env.HR_SOURCE_DATA_FILE);
-    console.log('Ready!');
+    console.log('Ready to Serve Hero Realms Cards and Data!  Beep boop, beep boop!');
 });
 
 // run this code when a message is received
 client.on('messageCreate', async message => {
     if (message.author.bot) return;
-    // 
     const matches = message.content.matchAll(regex);
+
     if (matches) {
         try {
+            var matchCount = 0;
             for(const match of matches) {
-                const searchTerm = match[1];
-                const results = await this.gallery.searchGallery(searchTerm, MAX_RESULTS);
-                if (results.length === 0) {
-                    await message.reply(`No cards found with name "${searchTerm}"`);
-                    continue;
+                //Limit the amount of SPAM to the channel
+                matchCount++;
+                if (matchCount > MAX_RESPONSES){
+                    message.reply(`You added too many searches to your message.  Please limit the number of card search requests for each message to ${MAX_RESPONSES}`);
+                    return; //This will break out of the loop and function, might not be the best approach, but it's the one we can take here.
                 }
-                if (results.length > 1) {
-                    await message.reply({content: `Multiple matches found for name "${searchTerm}", please select your choice:`, components: [await generateSelectMenu(this.gallery, results)]});
-                    continue;
+                const commandInput = match[1];
+                //console.log(`Command Input: "${commandInput}"`);
+                //Perform a better regex check here for data sanitization on searchTerm
+                //  - a-zA-Z (),|-!' - (space is included)
+
+                //Perform split on : if possible?
+
+                //[[card]] should show the default view
+                //[[t:card]] should show the text, no image
+                //[[i:card]] should only the image, nothing else
+                //[[v:card]] should show verbose mode (all the things)
+                //[a:ping], [a:uptime], [a:stats] -- administrative functions
+                //[a:days] -- days since the kickstarter for HR successfully funded
+                var cardSearch = true;
+                //"Default View"
+                var showText = false;
+                var showImage = true;
+                var showType = true;
+                var showCost = true;
+                var showFaction = true;
+                var showDefense = true;
```

Comment:
I think I'd prefer to only have the Type shown by default, since cost, faction and defense are a lot easier to read from the image.  The reason I originally added Type as a text field was that its font is pretty small in the card image and hard to read.

```suggestion
                var showCost = false;
                var showFaction = false;
                var showDefense = false;
```

Reply:
I added these to provide more levers based on the type of query for future enhancements (including full verbose card search, and a search that doesn't show the image of the card).  The changes for the defaults are good with me.